packages/malloy/src/lang/parse-malloy.ts

```diff
@@ -505,6 +514,37 @@ class MetadataStep implements TranslationStep {
   }
 }
 
+class CompletionsStep implements TranslationStep {
```

Comment:
@mtoy-googly-moogly Can you take a look at this and tell me whether this is the way you'd like me to do this? Tried to fit into your `TranslationStep` world as best as possible, though what I've done is a little like fitting an octagon into a circle — it works, but it feels a little funny doing it. Specifically, this is a request that could be repeated with different inputs (though in practice it's not, a new parse is created for each completions request currently).

Reply:
We talked in chat, but for the record 👍  -- You did it exactly as I would hope someone would coming into this new design. Nice work, glad it wasn't too much of a bother.

Reply:
also ...OMG COMPLETIONS 💯